src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -814,5 +899,33 @@ private void RecordBotSegment(string directoryPath, long tickNumber, byte[] json
                 RGDebug.LogException(e, $"ERROR: Unable to record JSON bot_segment for tick # {tickNumber}");
             }
         }
+
+        private void RecordLogs(string directoryPath, long tickNumber, string logs)
+        {
+            try
+            {
+                if (logs.Length == 0)
+                {
+                    return;
+                }
+                
+                // append logs to jsonl file
+                var path = $"{directoryPath}/logs.jsonl";
+                var fileWriteTask = File.AppendAllTextAsync(path, logs, _tokenSource.Token);
+                fileWriteTask.ContinueWith((nextTask) =>
+                {
+                    if (nextTask.Exception != null)
+                    {
+                        RGDebug.LogException(nextTask.Exception, $"ERROR: Unable to record logs for tick # {tickNumber}");
+                    }
+
+                });
+                _fileWriteTasks.Add((path,fileWriteTask));
+            }
+            catch (Exception e)
+            {
+                RGDebug.LogException(e, $"ERROR: Unable to record logs for tick # {tickNumber}");
+            }
```

Comment:
Do we need to worry about ticks attempting to write to this file at the same time? And additionally, logs from ticks being added in the incorrect order? I am honestly not sure

Reply:
Ooh I'm not sure either. Need to read up on how tasks work with I/O...

Reply:
This approach definitely seems unsafe. What's interesting is that Ashley's original implementation didn't account for this issue, unless I'm missing something from that PR. Maybe our logs were so sparse that the Tasks completed quickly enough that we never hit a case of concurrent / out-of-order write operations? 

Either way, I've pushed a change that moves ownership of I/O to the `LoggingObserver` itself. Since the timestamp is recorded as logs come in, we shouldn't need to write to file every single tick - this will now wait for the previous write task to complete before queuing a new one. This works in BossRoom, but I haven't gotten to try this out with <partner project> yet (it produces so many logs that it should be great for testing)

Reply:
Also leaving this open so Zack has some context on the idea and can give a second pair of eyes on this potential issue.